### PR TITLE
PP-2984 Add merchant telephone number

### PR DESCRIPTION
**Pull request by DanailMinchev**
## WHAT

Add merchant telephone number property on DAO and API level.
The `telephoneNumber` is optional.
The validation of the format of `telephoneNumber` is done in the client historically.

### CODE REVIEW

File: src/test/java/uk/gov/pay/adminusers/persistence/dao/ServiceDaoTest.java [hunk 1]
```diff
@@ -178,7 +184,7 @@ public void shouldGetRoleCountForAService() throws Exception {
 
         Long count = serviceDao.countOfUsersWithRoleForService(serviceExternalId, roleId);
 
-        assertThat(count, is(3l));
+        assertThat(count, is(3L));
```
Comment (alexbishop1) [inline src/test/java/uk/gov/pay/adminusers/persistence/dao/ServiceDaoTest.java]: 👍

File: src/test/java/uk/gov/pay/adminusers/resources/ServiceResourceUpdateMerchantDetailsTest.java [hunk 2]
```diff
@@ -82,6 +86,7 @@ public void shouldFail_whenUpdatingMerchantDetails_withoutMandatoryFields() thro
         Service service = Service.from(randomInt(), serviceExternalId, "existing-name");
         databaseHelper.addService(service, randomInt().toString());
         Map<String, Object> payload = ImmutableMap.of(
+                "telephone_number", "07700900000",
```
Comment (alexbishop1) [inline src/test/java/uk/gov/pay/adminusers/resources/ServiceResourceUpdateMerchantDetailsTest.java]: I know it was like this when you got here but this test would be better if it tested what happens if you miss just one mandatory field — right now it tests what happens when you miss all of them, which doesn’t feel as useful

**Comment by alexbishop1** [comment]
Oh, and the ticket number is wrong — it should be PP-2984

File: src/main/java/uk/gov/pay/adminusers/service/ServiceUpdater.java [hunk 3]
```diff
@@ -50,6 +51,12 @@ public Service doUpdateMerchantDetails(String serviceExternalId, UpdateMerchantD
         return serviceDao.findByExternalId(serviceExternalId)
                 .map(serviceEntity -> {
                     MerchantDetailsEntity merchantEntity = MerchantDetailsEntity.from(updateMerchantDetailsRequest);
+                    // in case we already set telephone number before we do not want to delete it
+                    // usually this condition may occur when we had direct debit gateway account and it was deleted
```
[the remark this answers is not in the document]
Comment (DanailMinchev) [reply src/main/java/uk/gov/pay/adminusers/service/ServiceUpdater.java]: as discussed I will remove this case

File: src/main/java/uk/gov/pay/adminusers/service/ServiceUpdater.java [hunk 4]
```diff
@@ -50,6 +51,12 @@ public Service doUpdateMerchantDetails(String serviceExternalId, UpdateMerchantD
         return serviceDao.findByExternalId(serviceExternalId)
                 .map(serviceEntity -> {
                     MerchantDetailsEntity merchantEntity = MerchantDetailsEntity.from(updateMerchantDetailsRequest);
```
[the remark this answers is not in the document]
Comment (DanailMinchev) [reply src/main/java/uk/gov/pay/adminusers/service/ServiceUpdater.java]: I removed the related logic and as discussed we now prefer the original naming

File: src/main/java/uk/gov/pay/adminusers/service/ServiceUpdater.java [hunk 5]
```diff
@@ -50,6 +51,12 @@ public Service doUpdateMerchantDetails(String serviceExternalId, UpdateMerchantD
...
+                    // in case we already set telephone number before we do not want to delete it
+                    // usually this condition may occur when we had direct debit gateway account and it was deleted
+                    if (((merchantEntity.getTelephoneNumber() == null) || (merchantEntity.getTelephoneNumber().isEmpty())) &&
```
[the remark this answers is not in the document]
Comment (DanailMinchev) [reply src/main/java/uk/gov/pay/adminusers/service/ServiceUpdater.java]: removed the logic

File: src/main/java/uk/gov/pay/adminusers/service/ServiceUpdater.java [hunk 6]
```diff
@@ -50,6 +51,12 @@ public Service doUpdateMerchantDetails(String serviceExternalId, UpdateMerchantD
...
+                            (serviceEntity.getMerchantDetailsEntity() != null)) {
+                        merchantEntity.setTelephoneNumber(serviceEntity.getMerchantDetailsEntity().getTelephoneNumber());
```
[the remark this answers is not in the document]
Comment (DanailMinchev) [reply src/main/java/uk/gov/pay/adminusers/service/ServiceUpdater.java]: removed the logic

File: src/test/java/uk/gov/pay/adminusers/model/UpdateMerchantDetailsRequestTest.java [hunk 7]
```diff
@@ -14,6 +14,7 @@
 public class UpdateMerchantDetailsRequestTest {
 
     private final String name = "name";
+    private final String telephoneNumber = "07700900000";
```
[the remark this answers is not in the document]
Comment (DanailMinchev) [reply src/test/java/uk/gov/pay/adminusers/model/UpdateMerchantDetailsRequestTest.java]: Yes, I think landline is better, I changed it to `03069990000`, thanks!

**Pull request merged**